spec/integration/email_outbound_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# We already have Email::Sender and Email::MessageBuilder specs along
+# with mailer specific mailer specs like UserEmail, but sometimes we need
+# to test things along the whole outbound flow including the MessageBuilder
+# and the Sender.
+describe "Outbound Email" do
+  def send_email(opts = {})
+    message = TestMailer.send_test("test@test.com", opts)
+    result = Email::Sender.new(message, :test_message).send
+    [message, result]
+  end
+
+  context "email custom headers" do
+    it "discards the custom header if it is one that we have already set based on arguments" do
```

Comment:
```suggestion
    it "discards the custom header if it is one that has already been set based on arguments" do
```

I don't think we usually use the first person perspective in our specs description.